### PR TITLE
Feature #12793: Added Exciting/Intensity/Nausea sorts on Ride List

### PR DESCRIPTION
**Pull request by gsckoco**
Based on #12793 
image
image

### CODE REVIEW

File: data/language/en-GB.txt [hunk 1]
```diff
@@ -3642,7 +3642,15 @@ STR_6383    :Open download page
...
-
+STR_6387    :Excitement rating: {COMMA2DP32}
+STR_6388    :Excitement rating: Not yet available
+STR_6389    :Intensity rating: {COMMA2DP32}
+STR_6390    :Intensity rating: Not yet available
+STR_6391    :Nausea rating: {COMMA2DP32}
+STR_6392    :Nausea rating: Not yet available
+STR_6393    :Excitement
+STR_6394    :Intensity
+STR_6395    :Nausea
```
Comment (Broxzier) [inline data/language/en-GB.txt:3653]: Please keep the empty line after this block.

**Comment by AaronVanGeffen** [comment]
Hmm, this seems to be partially duplicating existing strings:

```
STR_1473    :{WINDOW_COLOUR_2}Excitement rating: {BLACK}{COMMA2DP32}  ({STRINGID})
STR_1474    :{WINDOW_COLOUR_2}Excitement rating: {BLACK}Not yet available
STR_1475    :{WINDOW_COLOUR_2}Intensity rating: {BLACK}{COMMA2DP32}  ({STRINGID})
STR_1476    :{WINDOW_COLOUR_2}Intensity rating: {BLACK}Not yet available
STR_1477    :{WINDOW_COLOUR_2}Intensity rating: {OUTLINE}{RED}{COMMA2DP32}  ({STRINGID})
STR_1478    :{WINDOW_COLOUR_2}Nausea rating: {BLACK}{COMMA2DP32}  ({STRINGID})
STR_1479    :{WINDOW_COLOUR_2}Nausea rating: {BLACK}Not yet available
```

I wonder if it would be worth making the existing ones more flexible instead, e.g. by refactoring out the colours.

**Comment by gsckoco** [comment]
> Hmm, this seems to be partially duplicating existing strings:
> ...
> I wonder if it would be worth making the existing ones more flexible instead, e.g. by refactoring out the colours.
Or would it be better to write a function that strips a string of its colour values?

**Comment by AaronVanGeffen** [comment]
That sounds like allocating strings just for rendering? Unfortunately, the UI is redrawn every tick, so that may have a bit of an impact. In the current framework, it's easier to draw a string with e.g. two substrings. In this case, we'd use one string for the formatting, and have it process two strings by their string ids from passed format arguments.

**Comment by gsckoco** [comment]
> That sounds like allocating strings just for rendering? Unfortunately, the UI is redrawn every tick, so that may have a bit of an impact. In the current framework, it's easier to draw a string with e.g. two substrings. In this case, we'd use one string for the formatting, and have it process two strings by their string ids from passed format arguments.

Good point, instead of splitting the strings into format strings and text strings. Would it be possible to instead pass an optional bool into whatever function handles setting the colour, so instead of having 2 strings for rendering just have 1 string and have it ignore rendering the colour.

**Comment by AaronVanGeffen** [comment]
I see three colour variations and three labels. Something like:

1. `{WINDOW_COLOUR_2}{STRINGID}: {BLACK}{STRINGID}`
2. `{WINDOW_COLOUR_2}{STRINGID}: {OUTLINE}{RED}{STRINGID}`
3. `{STRINGID}: {STRINGID}` (introduced by this PR)

The first string id is the label, while the latter is either of:
* `{COMMA2DP32}  ({STRINGID})` (with the string id here being another label for the intensity)
* `{COMMA2DP32}` (variant introduced by this PR, but probably already exists)
* `Not yet available`

Perhaps I'm forgetting something, but it looks doable to me :)

**Comment by duncanspumpkin** [comment]
Do we not have the ability to pass the colour in as an argument?

**Comment by AaronVanGeffen** [comment]
We do, but (variable) window colours are not supported in this way. That, and there are two colours at play here.

**Comment by gsckoco** [comment]
Yeah, the ability is there, but if there is a tag that overwrites it it becomes useless. 
image
This is using `STR_EXCITEMENT_RATING` and the colour is being set by the `{WINDOW_COLOUR_2}` tag
This is what it should look like
image

**Comment by tupaschoal** [comment]
So where do we go from here? Is this on hold @AaronVanGeffen ?

**Comment by gsckoco** [comment]
Yeah, I think this is on hold until the template strings are sorted

**Comment by Gymnasiast** [comment]
@gsckoco When do you plan to make the changes Aaron requested?

**Comment by gsckoco** [comment]
@Gymnasiast not sure, I have been busy with college and not had much time to work on any projects,

**Comment by Gymnasiast** [comment]
Closing this as it has not been updated in two months. Let us know when/if you want to continue work and we can reopen it or open a new one.

**Comment by IntelOrca** [comment]
Probably best left to first refactoring the window to a class then re-visiting #9708

**Pull request closed without merging**